### PR TITLE
Scoped bindings

### PR DESCRIPTION
**Pull request by lionel-**
Useful to overscope symbols in a given environment instead of a child. I.e. when evaluating a stack-sensitive expression.

Also adds `env_get_list()` to retrieve multiple objects in a list.

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -383,6 +384,7 @@ export(syms)
 export(type_of)
 export(warn)
 export(warning_cnd)
+export(with_bindings)
```
Comment (hadley) [inline NAMESPACE]: I think you forgot to export `scoped_bindings`

File: R/env.R [hunk 2]
```diff
@@ -637,6 +637,61 @@ env_bind_fns <- function(.env, ...) {
...
+#' @return `scoped_bindings()` returns the values of old bindings
+#'   invisibly; `with_bindings()` returns the value of `expr`.
+#' @examples
+#' foo <- "foo"
+#' bar <- "bar"
+#'
+#' # `foo` will be temporarily rebinded while executing `expr`
+#' with_bindings(paste(foo, bar), foo = "rebinded")
+#' paste(foo, bar)
+scoped_bindings <- function(..., .env = .frame, .frame = caller_env()) {
```
Comment (hadley) [inline R/env.R]: Where does `.frame` come from?

File: R/env.R [hunk 3]
```diff
@@ -778,6 +835,12 @@ env_has <- function(env = caller_env(), nms, inherit = FALSE) {
 env_get <- function(env = caller_env(), nm, inherit = FALSE) {
   get(nm, envir = get_env(env), inherits = inherit)
 }
+#' @rdname env_get
+#' @export
+env_get_list <- function(env = caller_env(), nms, inherit = FALSE) {
```
Comment (hadley) [inline R/env.R]: `env_get_many()`?

File: R/env.R (hunk 2 above)
Comment (lionel-) [reply R/env.R]: It's the third argument with default `caller_env()`.

File: R/env.R (hunk 3 above)
Comment (lionel-) [reply R/env.R]: I thought `_list` would be consistent with the return type rule. We already have other `_list` like `dots_list()`.

File: R/env.R (hunk 2 above)
Comment (hadley) [reply R/env.R]: Doh

File: R/env.R (hunk 3 above)
Comment (hadley) [reply R/env.R]: Ok

**Pull request merged**